src/content/guides/progressive-web-application.md

```diff
@@ -0,0 +1,150 @@
+
+This is what we aim to change. Once we reach the end of this module we should be able to stop the server, hit refresh and still see our application.
+
+## Adding Workbox
+
+Let's add the Workbox webpack plugin and adjust the `webpack.config.js` file:
+
+``` bash
+npm install workbox-webpack-plugin --save-dev
+```
+
+__webpack.config.js__
+
+``` diff
+  const path = require('path');
+  const HtmlWebpackPlugin = require('html-webpack-plugin');
+  const CleanWebpackPlugin = require('clean-webpack-plugin');
++ const WorkboxPlugin = require('workbox-webpack-plugin');
+
+  module.exports = {
+    entry: {
+      app: './src/index.js',
+      print: './src/print.js'
+    },
+    plugins: [
+        new CleanWebpackPlugin(['dist']),
+        new HtmlWebpackPlugin({
+            title: 'Output Management'
+-       })
++       }),
++       new WorkboxPlugin({
++           // these options encourage the ServiceWorkers to get in there fast 
++           // and not allow any straggling "old" SWs to hang around
++           clientsClaim: true,
++           skipWaiting: true,
++       })
+    ],
+    output: {
+      filename: '[name].bundle.js',
+      path: path.resolve(__dirname, 'dist')
+    }
+  };
+```
+
+With that in place, let's see what happens when we do an `npm run build`:
+
+``` bash
+clean-webpack-plugin: /mnt/c/Source/webpack-follow-along/dist has been removed.
+Hash: 6588e31715d9be04be25
+Version: webpack 3.10.0
+Time: 782ms
+                                                Asset       Size  Chunks                    Chunk Names
+                                        app.bundle.js     545 kB    0, 1  [emitted]  [big]  app
+                                      print.bundle.js    2.74 kB       1  [emitted]         print
+                                           index.html  254 bytes          [emitted]
+precache-manifest.b5ca1c555e832d6fbf9462efd29d27eb.js  268 bytes          [emitted]
+                                                sw.js       1 kB          [emitted]
+   [0] ./src/print.js 87 bytes {0} {1} [built]
+   [1] ./src/index.js 477 bytes {0} [built]
+   [3] (webpack)/buildin/global.js 509 bytes {0} [built]
+   [4] (webpack)/buildin/module.js 517 bytes {0} [built]
+    + 1 hidden module
+Child html-webpack-plugin for "index.html":
+     1 asset
+       [2] (webpack)/buildin/global.js 509 bytes {0} [built]
+       [3] (webpack)/buildin/module.js 517 bytes {0} [built]
+        + 2 hidden modules
+```
+
+As you can see, we now have 2 extra files being generated; `sw.js` and the more verbose `precache-manifest.b5ca1c555e832d6fbf9462efd29d27eb.js`. `sw.js` is the Service Worker file and `precache-manifest.b5ca1c555e832d6fbf9462efd29d27eb.js` is a file that `sw.js` imports requires so it can run. Your own generated files will likely be different; but you should have an `sw.js` file there.
+
+So we're now at the happy point of having produced a Service Worker. What to do with it?
```

Comment:
I might simplify the last bit to `What's next?` instead of `What to do with it?`.

Reply:
done.